.github/workflows/deploy_production.yml

```
@@ -0,0 +1,154 @@
+
+  deploy_ios_production:
+    runs-on: macos-latest
+    environment: production
+    defaults:
+      run:
+        working-directory: ./sample/ios
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Cache Gradle
+        uses: actions/cache@v3.3.2
+        with:
+          path: |
+            ~/.gradle/caches/modules-*
+            ~/.gradle/caches/jars-*
+            ~/.gradle/caches/build-cache-*
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Install SSH key - For accessing match repo
+        uses: webfactory/ssh-agent@v0.7.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+      - uses: actions/cache@v3
+        id: bunlderCache
+        with:
+          path: vendor/bundle
+          key: ${{ runner.os }}-gems-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-gems-
+
+      - name: Bundle install
+        run: bundle install --path vendor/bundle
+
+      - name: Cache Pods
+        uses: actions/cache@v2
+        id: cocoapodCache
+        with:
+          path: Pods
+          key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-pods-
+
+      - name: Install Pods Dependencies
+        run: bundle exec pod install
+        shell: bash
+
+      - name: Decode buildKonfig properties
+        run: |
+          cd ..
+          echo ${{ secrets.BUILD_KONFIG_PROPERTIES }} | base64 --decode > buildKonfig.properties
+
+      - name: Paste Google-Info.plist from .secrets
+        run: |
+          mkdir -p sample/Configurations/Plists/GoogleService/Staging
+          echo ${{ secrets.IOS_GOOGLE_INFO_PLIST }} | base64 --decode > ./sample/Configurations/Plists/GoogleService/Staging/GoogleService-Info.plist
+
+      - name: Paste Constants for Fastlane from .secrets
+        run: |
+            mkdir -p fastlane/Constants
+            echo ${{ secrets.IOS_FASTLANE_CONSTANTS }} | base64 --decode > ./fastlane/Constants/Constant.swift
+
+      - name: Generate KMM frameworks for Cocoapods
+        run: |
+          cd ..
+          ./gradlew generateDummyFramework
+
+      # - name: Build and Test
+      #   run: bundle exec fastlane buildAndTest
+
+      - name: Match AppStore
+        run: bundle exec fastlane syncAppStoreCodeSigning
+        env:
+          MATCH_PASSWORD: ${{ secrets.MATCH_PASS }}
+
+      - name: Build App and Distribute to AppStore
+        run: bundle exec fastlane buildAndUploadToAppStore
+        env:
+          APPSTORE_CONNECT_API_KEY: ${{ secrets.APPSTORE_CONNECT_API_KEY }}
+          API_KEY_ID: ${{ secrets.API_KEY_ID }}
+          ISSUER_ID: ${{ secrets.ISSUER_ID }}
```

Since we have 2 platforms, which are IOS and Android, It would be neater if we had a prefix to every separate secret variable. This will help make them easier to maintain (updating/removing/grouping). What do you think?

```suggestion
          APPSTORE_CONNECT_API_KEY: ${{ secrets.IOS_APPSTORE_CONNECT_API_KEY }}
          API_KEY_ID: ${{ secrets.IOS_API_KEY_ID }}
          ISSUER_ID: ${{ secrets.IOS_ISSUER_ID }}
```